internal/services/datafactory/data_factory_resource.go

```diff
@@ -102,6 +102,11 @@ func resourceDataFactory() *pluginsdk.Resource {
 							Required:     true,
 							ValidateFunc: validation.StringIsNotEmpty,
 						},
+						"publish_enabled": {
```

Comment:
Can we rename this to 
```suggestion
						"publishing_enabled": {
```

Reply:
Sure, I have updated this PR.